File: gin-vue-admin-main/web/src/api/medicine/mtDrugTypeStair.js
import service from '@/utils/request'

// 获取所有一级分类
export const getAllMtDrugTypeStair = () => {
  return service({
    url: '/mtDrugTypeStair/getMtDrugTypeStairPublic',
    method: 'get'
  })
}

// 分页获取一级分类列表
export const getMtDrugTypeStairList = (params) => {
  return service({
    url: '/mtDrugTypeStair/getMtDrugTypeStairList',
    method: 'get',
    params
  })
}

// 创建一级分类
export const createMtDrugTypeStair = (data) => {
  return service({
    url: '/mtDrugTypeStair/createMtDrugTypeStair',
    method: 'post',
    data
  })
}

// 更新一级分类
export const updateMtDrugTypeStair = (data) => {
  return service({
    url: '/mtDrugTypeStair/updateMtDrugTypeStair',
    method: 'put',
    data
  })
}

// 删除一级分类
export const deleteMtDrugTypeStair = (params) => {
  return service({
    url: '/mtDrugTypeStair/deleteMtDrugTypeStair',
    method: 'delete',
    params
  })
}

// 批量删除一级分类
export const deleteMtDrugTypeStairByIds = (params) => {
  return service({
    url: '/mtDrugTypeStair/deleteMtDrugTypeStairByIds',
    method: 'delete',
    params
  })
}

// 根据ID查询一级分类
export const findMtDrugTypeStair = (params) => {
  return service({
    url: '/mtDrugTypeStair/findMtDrugTypeStair',
    method: 'get',
    params
  })
} 

File: gin-vue-admin-main/server/router/medicine/mt_drug_type_stair.go
package medicine

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type MtDrugTypeStairRouter struct{}

// InitMtDrugTypeStairRouter 初始化 mtDrugTypeStair表 路由信息
func (s *MtDrugTypeStairRouter) InitMtDrugTypeStairRouter(Router *gin.RouterGroup, PublicRouter *gin.RouterGroup) {
	mtDrugTypeStairRouter := Router.Group("mtDrugTypeStair").Use(middleware.OperationRecord())
	mtDrugTypeStairRouterWithoutRecord := Router.Group("mtDrugTypeStair")
	mtDrugTypeStairRouterWithoutAuth := PublicRouter.Group("mtDrugTypeStair")
	{
		mtDrugTypeStairRouter.POST("createMtDrugTypeStair", mtDrugTypeStairApi.CreateMtDrugTypeStair)             // 新建mtDrugTypeStair表
		mtDrugTypeStairRouter.DELETE("deleteMtDrugTypeStair", mtDrugTypeStairApi.DeleteMtDrugTypeStair)           // 删除mtDrugTypeStair表
		mtDrugTypeStairRouter.DELETE("deleteMtDrugTypeStairByIds", mtDrugTypeStairApi.DeleteMtDrugTypeStairByIds) // 批量删除mtDrugTypeStair表
		mtDrugTypeStairRouter.PUT("updateMtDrugTypeStair", mtDrugTypeStairApi.UpdateMtDrugTypeStair)              // 更新mtDrugTypeStair表
	}
	{
		mtDrugTypeStairRouterWithoutRecord.GET("findMtDrugTypeStair", mtDrugTypeStairApi.FindMtDrugTypeStair)       // 根据ID获取mtDrugTypeStair表
		mtDrugTypeStairRouterWithoutRecord.GET("getMtDrugTypeStairList", mtDrugTypeStairApi.GetMtDrugTypeStairList) // 获取mtDrugTypeStair表列表
		mtDrugTypeStairRouterWithoutRecord.GET("getAllMtDrugTypeStair", mtDrugTypeStairApi.GetAllMtDrugTypeStair)   // 获取所有一级分类
	}
	{
		mtDrugTypeStairRouterWithoutAuth.GET("getMtDrugTypeStairPublic", mtDrugTypeStairApi.GetAllMtDrugTypeStair) // mtDrugTypeStair表开放接口
	}
}
